providers/job.rb

```diff
@@ -9,49 +9,36 @@ def load_current_resource
   @current_resource.project(@new_resource.project)
 end
 
-def stringify(h)
-  case h
-  when Hash
-    h.inject({}) do |memo, (k, v)|
-      memo[k.to_s] = stringify v
-      memo
-    end
-  when Array
-    h.map { |el| stringify el }
-  else
-    h
-  end
-end
-
 action :create do
   require 'rundeck'
   require 'yaml'
 
   client = Rundeck.client(endpoint: @new_resource.endpoint, api_token: @new_resource.api_token)
+  job = get_job(client, @current_resource.project, @current_resource.name)
 
-  job = find_job client, @current_resource.project, @current_resource.name
-
-  updated_job = stringify @new_resource.config.dup
+  updated_job = stringify(@new_resource.config.dup)
 
   # hydrate updated_job
   updated_job['name']    ||= @current_resource.name
   updated_job['project'] ||= @current_resource.project
   updated_job['uuid']    ||= job['uuid'] if job
 
-  tweak_existing job if job
+  # we need to specify uuid, not id to be able to update
+  job.delete('id') if job
 
   action_name = 'create'
   action_name = 'update' if job
 
   if job.nil? || !equal(job, updated_job)
     Chef::Log.debug('before: ' + job.inspect)
-    Chef::Log.debug('after: ' + updated_job.inspect)
+    Chef::Log.debug(' after: ' + updated_job.inspect)
 
     # yamlize update_job
     job_yaml = [updated_job].to_yaml.gsub(/^---\n/, '')
 
     converge_by "#{action_name} job #{@current_resource.project}/#{@current_resource.name}" do
-      response = client.import_jobs(job_yaml, 'yaml', opts)
+      # dupeOption allow us to update jobs (default is create, which fails)
+      response = client.import_jobs(job_yaml, 'yaml', opts('dupeOption' => 'update'))
```

Comment:
`opts.merge('query' => {'dupeOption' => 'update'})` would remove the need for adding arguments to opts method